### PR TITLE
server:Add game event handler for JOIN_GAME event

### PR DESCRIPTION
**Pull request by ritwik-69**
# Fixes: #93 

## Description
Created the event handler for JOIN_GAME event to let new players join the server before the game is started by the host.

## Checklist
- [x] I have tested these changes locally.
- [x] I have reviewed the code and ensured it follows the project's coding guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have assigned reviewers to this pull request.

### CODE REVIEW

File: backend/src/uno-game-engine/events/joinGame.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { assert } from 'console';
+import { GameEngine } from '../engine';
+import { registerEventHandler } from '../gameEvents';
+
+export function joinGame(game: GameEngine, event: GameEvent): EventResult {
+    assert(event.type === 'JOIN_GAME', 'Invalid event type');
+    const player: Player = { id: event.playerId, cards: [] };
+    game.addPlayer(player);
```
Comment (kuv2707) [inline backend/src/uno-game-engine/events/joinGame.ts:8]: We should allot the cards to the currently joining player too.

**Review by kuv2707 — COMMENTED** [review]
LGTM!
Left a suggestion

File: backend/src/uno-game-engine/events/joinGame.ts (hunk 1 above)
Comment (kuv2707) [reply backend/src/uno-game-engine/events/joinGame.ts:8]: This would require tweaking the `allotCards` function in the engine module.

Comment (ritwik-69) [reply backend/src/uno-game-engine/events/joinGame.ts:8]: @kuv2707 I Thought that would be done after all the players have joined and the status have been changed to started?

Comment (ritwik-69) [reply backend/src/uno-game-engine/events/joinGame.ts:8]: @kuv2707 what should i tweak about the `allotCards` function

**Pull request merged**